### PR TITLE
New air alarm sprite take two

### PR DESCRIPTION
**Pull request by MTandi**
dreamseeker_3knwrXrSzO

image

## About The Pull Request

Reopen of #74932 

Resprites the air alarm and adds a new sprite that will be needed for an air alarm that is remotely connected to a sensor via #75187 

Also fixed air alarm not checking air status on power change.

## Why It's Good For The Game

Newer, more compact sprite. 

It will make it easier for map makers to stack remotely connected air alarms on a tile to make an air monitoring wall.

## Changelog

:cl:
image: New air alarm sprite
fix: Air alarm properly updates status when powered
/:cl:

### CODE REVIEW

**Comment by zxaber** [comment]
Does this still make it so north-facing air alarms no longer flip?

**Comment by MTandi** [comment]
> Does this still make it so north-facing air alarms no longer flip?

Yes, like APC.

image

**Comment by zxaber** [comment]
Still heavily dislike that. I dislike it for APCs too, but they're close to being square so the difference is minor.

Rotating Air Alarms is super helpful when you can see multiple rooms and need to find the Air Alarm to *this* room because oh fuck it's siphoning and people are gonna die and also they're gonna blame you because EMPs are a bitch.

Considering how unique the sprite is (like the old sprite), I can't really see a good argument *against* the rotation effect.

**Comment by MTandi** [comment]
> Still heavily dislike that. I dislike it for APCs too, but they're close to being square so the difference is minor.
> 
> Rotating Air Alarms is super helpful when you can see multiple rooms and need to find the Air Alarm to _this_ room because oh fuck it's siphoning and people are gonna die and also they're gonna blame you because EMPs are a bitch.
> 
> Considering how unique the sprite is (like the old sprite), I can't really see a good argument _against_ the rotation effect.

I did it for consistency, because it looks weird when supply request consoles, APCs, fire extinguisher cabinets and most of other wall equipment is not rotated, while air alarm is. Also it's bad to rotate something that has text.

It may be true that the directional one works better in this particular edge case of a AI role, but I don't think that it justifies to add 4 times more sprites and make things look worse for everyone else.

Also the Wallening is comming, this sprite will be replaced and you will have even more issues finding and clicking that air alarm. So this is mostly a temporary thing or for downstreams that won't accept Wallening.

**Comment by Dorsisdwarf** [comment]
Amazing looking sprite, a perfect update to an old icon that was more than showing its age

**Comment by potato1234x** [comment]
this sprite is eye candy

**Comment by NotSoEpikName** [comment]
no way TG Sriters can do realy good shit unstead of bullshit?
imposible
yet still hey good work i like the sprites

File: code/modules/atmospherics/machinery/air_alarm/_air_alarm.dm [hunk 1]
```diff
@@ -122,6 +122,12 @@ GLOBAL_LIST_EMPTY_TYPED(air_alarms, /obj/machinery/airalarm)
 	GLOB.air_alarms -= src
 	return ..()
 
+/obj/machinery/airalarm/power_change()
+	var/turf/our_turf = connected_sensor ? get_turf(connected_sensor) : get_turf(src)
+	var/datum/gas_mixture/environment = our_turf.return_air()
+	check_danger(our_turf, environment, environment.temperature)
+	. = ..()
```
Comment (san7890) [inline code/modules/atmospherics/machinery/air_alarm/_air_alarm.dm:129]:
```suggestion
	return ..()
```

**Review by san7890 — CHANGES_REQUESTED** [review]
damn this PR got buried, just one nit

**Comment by MTandi** [comment]
> damn this PR got buried, just one nit

Please merge it, I can't handle the urge to keep polishing my PRs while they're open for too long.

**Comment by zxaber** [comment]
Please make them rotate, I don't know I can deal with fire alarms take two.

**Comment by MTandi** [comment]
> Please make them rotate, I don't know I can deal with fire alarms take two.

Man, it's one of the main reasons for this PR. I made the sprite smaller and close to the edge of wall to make sure that you won't have issues with finding air alarm in the room. APC already follows the same logic and this being non-directional is for consistency.

And object overlapping is a general mapping issue, not related to air alarms. Mappers should avoid placing wall objects on the same tile, even on the opposite sides of the wall, and we need to resolve these separately.

Please try how this works and make a PR with directions if it really feels much worse.

**Comment by zxaber** [comment]
>Please try how this works and make a PR with directions if it really feels much worse.

Alright, this is fair. I will give it an honest trial.

**Comment by zkxs** [comment]
Given @zxaber's concerns with how this might look for AI or other players who can see multiple sides of a wall, I figured I'd load up MetaStation with this PR merged and see how things look as an AI. I've attached some screenshots of areas I found interesting due to their high air alarm density. 

<details>
<summary>A bunch of screenshots (click to expand)</summary>

You fool! Now my single comment takes up tremendous quantities of vertical space in the page!

EVA Storage

Virology

Permabrig

Freezer

Grav gen

</details>

Surprisingly, the side of the walls air alarms are on is much less ambiguous than I'd thought it would be, although in hindsight I shouldn't be because during my AI gameplay I don't struggle too much with APCs, and it's basically the same situation as this.

**Pull request merged**